### PR TITLE
Reformat all the things

### PR DESCRIPTION
**Pull request by djarek**
Added support for clang-format (```make clang-format```) and implemented integration with travis. Travis builds will now fail if the PR is improperly formatted.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,6 +20,8 @@ addons:
       - libmysqlclient-dev
       - libpugixml-dev
 before_script:
+  - ./verify-format.sh
   - mkdir build && cd build
   - cmake -DCMAKE_BUILD_TYPE=Release -DUSE_LUAJIT=${LUAJIT} ..
-script: make -j2
+script:
```
Comment (ranisalt) [inline .travis.yml]: Why?

File: CMakeLists.txt [hunk 2]
```diff
@@ -16,7 +16,6 @@ if (CMAKE_COMPILER_IS_GNUCXX)
 endif()
 
 include(FindCXX11)
-
```
Comment (ranisalt) [inline CMakeLists.txt]: Why?

File: cmake/ClangTools.cmake [hunk 3]
```diff
@@ -0,0 +1,10 @@
+find_program(CLANG_FORMAT NAMES "clang-format")
+
+if(CLANG_FORMAT)
+  message(STATUS "clang-format found: -i -style=file ${tfs_HDR}")
+  add_custom_target(
+    clang-format
+    COMMAND ${CLANG_FORMAT}
+    -i -style=file ${tfs_SRC} ${tfs_HDR}
+    )
+endif()
```
Comment (ranisalt) [inline cmake/ClangTools.cmake]: I don't think you should check for clang-format executable. Just let it fail if the executable doesn't exist.

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -75,4 +75,6 @@ set(tfs_SRC
 	${CMAKE_CURRENT_LIST_DIR}/weapons.cpp
 	${CMAKE_CURRENT_LIST_DIR}/wildcardtree.cpp
 	PARENT_SCOPE)
+file(GLOB_RECURSE tfs_HDR *.h)
```
Comment (ranisalt) [inline src/CMakeLists.txt]: If we use glob for headers, we should use for source files too, for the sake of consistency.

File: src/actions.cpp [hunk 5]
```diff
@@ -110,14 +109,16 @@ bool Actions::registerEvent(Event* event, const pugi::xml_node& node)
 
 		auto result = useItemMap.emplace(iterId, action);
 		if (!result.second) {
-			std::cout << "[Warning - Actions::registerEvent] Duplicate registered item with id: " << iterId << " in fromid: " << fromId << ", toid: " << toId << std::endl;
+			std::cout << "[Warning - Actions::registerEvent] Duplicate registered item with id: " << iterId << " in fromid: " << fromId << ", toid: " << toId
+			          << std::endl;
```
Comment (ranisalt) [inline src/actions.cpp]: SEXY!

File: src/actions.h [hunk 6]
```diff
@@ -28,68 +28,68 @@ using ActionFunction = std::function<bool(Player* player, Item* item, const Posi
...
-		virtual ReturnValue canExecuteAction(const Player* player, const Position& toPos);
-		virtual bool hasOwnErrorHandler() {
-			return false;
-		}
-		virtual Thing* getTarget(Player* player, Creature* targetCreature, const Position& toPosition, uint8_t toStackPos) const;
+	virtual ReturnValue canExecuteAction(const Player* player, const Position& toPos);
+	virtual bool hasOwnErrorHandler()
+	{
+		return false;
+	}
```
Comment (ranisalt) [inline src/actions.h]: Try to allow short functions as one-liners, it reduces clutter.

File: src/ban.h [hunk 7]
```diff
@@ -20,15 +20,18 @@
...
 };
 
-struct ConnectBlock {
-	constexpr ConnectBlock(uint64_t lastAttempt, uint64_t blockTime, uint32_t count) :
-		lastAttempt(lastAttempt), blockTime(blockTime), count(count) {}
+struct ConnectBlock
+{
+	constexpr ConnectBlock(uint64_t lastAttempt, uint64_t blockTime, uint32_t count) : lastAttempt(lastAttempt), blockTime(blockTime), count(count)
+	{
+	}
```
Comment (ranisalt) [inline src/ban.h]: :( this is so ugly

File: src/combat.cpp [hunk 8]
```diff
@@ -1280,21 +1267,11 @@ void AreaCombat::setupArea(int32_t length, int32_t spread)
...
-		{0, 0, 8, 7, 6, 5, 5, 5, 6, 7, 8, 0, 0},
-		{0, 0, 0, 8, 7, 6, 6, 6, 7, 8, 0, 0, 0},
-		{0, 0, 0, 0, 8, 8, 7, 8, 8, 0, 0, 0, 0},
-		{0, 0, 0, 0, 0, 0, 8, 0, 0, 0, 0, 0, 0}
-	};
+	int32_t area[13][13] = {{0, 0, 0, 0, 0, 0, 8, 0, 0, 0, 0, 0, 0}, {0, 0, 0, 0, 8, 8, 7, 8, 8, 0, 0, 0, 0}, {0, 0, 0, 8, 7, 6, 6, 6, 7, 8, 0, 0, 0},
+	                        {0, 0, 8, 7, 6, 5, 5, 5, 6, 7, 8, 0, 0}, {0, 8, 7, 6, 5, 4, 4, 4, 5, 6, 7, 8, 0}, {0, 8, 6, 5, 4, 3, 2, 3, 4, 5, 6, 8, 0},
+	                        {8, 7, 6, 5, 4, 2, 1, 2, 4, 5, 6, 7, 8}, {0, 8, 6, 5, 4, 3, 2, 3, 4, 5, 6, 8, 0}, {0, 8, 7, 6, 5, 4, 4, 4, 5, 6, 7, 8, 0},
+	                        {0, 0, 8, 7, 6, 5, 5, 5, 6, 7, 8, 0, 0}, {0, 0, 0, 8, 7, 6, 6, 6, 7, 8, 0, 0, 0}, {0, 0, 0, 0, 8, 8, 7, 8, 8, 0, 0, 0, 0},
+	                        {0, 0, 0, 0, 0, 0, 8, 0, 0, 0, 0, 0, 0}};
```
Comment (ranisalt) [inline src/combat.cpp]: This is pretty fucked up now

File: src/const.h [hunk 9]
```diff
@@ -341,10 +318,11 @@ enum TextColor_t : uint8_t {
 	TEXTCOLOR_NONE = 255,
 };
 
-enum Icons_t {
+enum Icons_t
+{
 	ICON_POISON = 1 << 0,
 	ICON_BURN = 1 << 1,
-	ICON_ENERGY =  1 << 2,
+	ICON_ENERGY = 1 << 2,
```
Comment (ranisalt) [inline src/const.h]: How come @WibbenZ didn't see it yet

File: src/ban.h (hunk 7 above)
Comment (djarek) [inline src/ban.h]: Reducing column limit would probably help here.

Comment (ranisalt) [inline src/ban.h]: I mean, it should be `constexpr ConnectBlock(uint64_t lastAttempt, uint64_t blockTime, uint32_t count) : lastAttempt(lastAttempt), blockTime(blockTime), count(count) {}` on a single line or `{}` on the next line.

File: src/actions.h (hunk 6 above)
Comment (djarek) [inline src/actions.h]: I'll set it so that inline function definitions will be one liners (if they fit).

File: src/combat.cpp (hunk 8 above)
Comment (djarek) [inline src/combat.cpp]: Should be better now.

File: src/CMakeLists.txt (hunk 4 above)
Comment (djarek) [inline src/CMakeLists.txt]: We use the glob only for clang-format (and clang-tidy in the future). Using it for the build is a bad idea (screws up cmake's change detection system AFAIK).

**Comment by Leo32onGIT** [comment]
Although its generic enough of an integration to have reference value, it'll just sit here forever.
The redundant branch is now clutter too, should be cleaned up _(make use of [readme](https://github.com/otland/forgottenserver/tree/clang_format))_ or removed.

**Pull request closed without merging**